core/src/main/java/org/apache/druid/common/exception/SanitizableException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.common.exception;
+
+import java.util.function.Function;
+
+public interface SanitizableException
```

Comment:
This shouldn't effect/change the UI/UX or configs/settings so I think if we want, we can easily rename it later

Reply:
javadocs please. Explanation about how this is expected to be used and why this class exists will be helpful.

Reply:
Instead of calling this an Exception - let's make this a Factory class since the interface can't extend an Exception.

```
public interface SanitizingExceptionFactory<T extends Exception>
{ 
    T transform(T exception, Function<String, String> errorMessageTransformFunction);
}
```

Reply:
> Instead of calling this an Exception - let's make this a Factory class since the interface can't extend an Exception

Please ignore this suggestion. It makes the `ErrorResponseTransformStrategy` clunky